app/models/project.rb

```diff
@@ -4,6 +4,7 @@ class Project < ApplicationRecord
     include Elasticsearch::Model::Callbacks
   end
 
+  # TODO: (gdingle): add counter_cache if we ever migrate user assoc to "has_many :through"
```

Comment:
I think you should change the comment to explain the limitation of `counter_cache` with `has_and_belongs_to_many`.
It is not a TODO per se...
Better keep the TODO tag for things we should do soon.

What was the best reference you found for this? I was looking and saw people complaining about has_many :through either.

Reply:
Another question, what would be the limitation of changing to has_many+through? Could we just change it now?

Reply:
I spent about 30min researching this issue and I found that `has_and_belongs_to_many` is not supported for counters and many people recommend against ever using `has_and_belongs_to_many` for other reasons. 

I thought about migrating to has_many+through but I felt it was too much for this PR and I'm not sure enough of other consequences.

Reply:
I understand. This is where test would give some comfort.
The "cheap" on the other hand is relative... since the number of analytics calls might increase rapidly. 
But I guess we can move on for now, and just keep an eye.